File: product-ui/src/app/product/product.ts


export class Product {
  
  id: string;
  name: string;
  price: number;
  pictureUrl: string;
  brand: string;
  category: string;
  description:string;
  quantity: number;
    
    constructor(id: string, name: string, brand:string, category: string, price: number, quantity: number, pictureUrl: string, description: string) {
      this.id = id;
      this.name = name;
      this.price = price;
      this.pictureUrl = pictureUrl;
      this.brand = brand;
      this.category = category;
      this.description = description;
      this.quantity = quantity;
    }
}
 

File: product-ms/src/main/java/com/example/productmicroservice/model/Product.java
package com.example.productmicroservice.model;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.TextIndexed;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;


@Data
@Document(collection = "ProductDB")
public class Product {

    @Id
	private String id;

    @TextIndexed
	private String name;
    
    @TextIndexed
	private String category;

    @TextIndexed
    private String brand;
    private Double price;
    private Double quantity;
    private String pictureUrl;

    @TextIndexed
    private String description;
}
